src/ecrc-frontend/src/components/page/consent/Consent.test.js

```diff
@@ -1,23 +1,92 @@
 import React from "react";
 import { create } from "react-test-renderer";
+import { Router } from "react-router-dom";
+import {
+  render,
+  fireEvent,
+  getAllByRole,
+  getByText
+} from "@testing-library/react";
+import { createMemoryHistory } from "history";
+import { generateJWTToken } from "../../../modules/AuthenticationHelper";
 
 import Consent from "./Consent";
 
 describe("Consent Page Component", () => {
+  window.scrollTo = jest.fn();
+
+  const header = {
+    name: "Criminal Record Check"
+  };
+
+  const page = {
+    header
+  };
+
+  beforeEach(() => {
+    sessionStorage.setItem("validator", "secret");
+    sessionStorage.setItem("uuid", "unique123");
+    generateJWTToken({
+      actionsPerformed: ["userConfirmation"],
+      authorities: ["Authorized"]
+    });
+  });
+
   test("Matches the snapshot", () => {
-    const header = {
-      name: "Criminal Record Check"
-    };
+    const consent = create(<Consent page={page} />);
+    expect(consent.toJSON()).toMatchSnapshot();
+  });
 
-    const page = {
-      header
-    };
+  test("Validate Cancel Redirect", () => {
+    const history = createMemoryHistory();
+    const { container } = render(
+      <Router history={history}>
+        <Consent page={page} />
+      </Router>
+    );
 
-    const onContinueClick = () => jest.fn();
+    fireEvent.click(getByText(container, "Cancel and Exit"));
+    expect(history.location.pathname).toEqual("/hosthome");
+  });
 
-    const consent = create(
-      <Consent page={page} onContinueClick={onContinueClick} />
+  test("Validate Continue Redirect", () => {
+    const history = createMemoryHistory();
+    const { container } = render(
+      <Router history={history}>
+        <Consent page={page} />
+      </Router>
     );
-    expect(consent.toJSON()).toMatchSnapshot();
+
+    expect(getByText(container, "Continue").disabled).toBeTruthy();
+
+    const checkbox = getAllByRole(container, "checkbox");
+
+    fireEvent.click(checkbox[0]);
+    fireEvent.click(checkbox[1]);
+    fireEvent.click(checkbox[2]);
+
+    expect(getByText(container, "Continue").disabled).toBeFalsy();
+
+    fireEvent.click(getByText(container, "Continue"));
+    expect(history.location.pathname).toEqual(
+      "/criminalrecordcheck/applicationform"
+    );
+  });
+
+  test("Validate Redirect to Home when unauthorized", () => {
```

Comment:
this is a really good test case, thank you! We need to add a similar test to all pages :)